bumble/sdp.py

```diff
@@ -167,7 +167,7 @@ class DataElement:
         UUID: lambda x: DataElement(
             DataElement.UUID, core.UUID.from_bytes(bytes(reversed(x)))
         ),
-        TEXT_STRING: lambda x: DataElement(DataElement.TEXT_STRING, x.decode('utf8')),
+        TEXT_STRING: lambda x: DataElement(DataElement.TEXT_STRING, x.decode('latin1')),
```

Comment:
After checking the spec, since no specific encoding is required for Data Element objects of type String, I think the best option here is to leave it as a byte array and then let the user of the object perform encoding or decoding, as appropriate depending on their context and profile (the encoding may be specified elsewhere for certain use cases)